src/extension/features/budget/subtract-upcoming-from-available/index.js

```diff
@@ -0,0 +1,124 @@
+import { Feature } from 'toolkit/extension/features/feature';
+import { isCurrentRouteBudgetPage } from 'toolkit/extension/utils/ynab';
+import { getEmberView } from 'toolkit/extension/utils/ember';
+import { formatCurrency } from 'toolkit/extension/utils/currency';
+import { addToolkitEmberHook } from 'toolkit/extension/utils/toolkit';
+
+let totalUpcoming = 0;
+
+export class SubtractUpcomingFromAvailable extends Feature {
+  shouldInvoke() {
+    return isCurrentRouteBudgetPage();
+  }
+
+  invoke() {
+    addToolkitEmberHook(this, 'budget-bar-v2', 'didRender', this.run);
+    addToolkitEmberHook(this, 'budget-bar-v2', 'didUpdate', this.run);
```

Comment:
I _think_ we should be able to just use `didRender` and remove this one, the difference between the two is that `didRender` is called on every render (including the first) and `didUpdate` is only called on renders after the first.

Also, I don't think `budget-bar-v2` is what we want. I think `budget-table-row` is what you're after. That will call your `this.run` function with the element of the row every time it renders.